android/src/main/java/com/rncamerakit/RNCameraKitModule.kt

```diff
@@ -62,13 +68,16 @@ class RNCameraKitModule(private val reactContext: ReactApplicationContext) : Rea
      * @param promise The promise to resolve the capture result.
      */
     @ReactMethod
-    fun capture(options: ReadableMap, viewTag: Int, promise: Promise) {
+    override fun capture(options: ReadableMap?, tag: Double?, promise: Promise) {
         // CameraManager does not allow us to return values
         val context = reactContext
-        val uiManager = context.getNativeModule(UIManagerModule::class.java)
-        context.runOnUiQueueThread {
-            val view = uiManager?.resolveView(viewTag) as CKCamera
-            view.capture(options.toHashMap(), promise)
+        val viewTag = tag?.toInt()
+        if (viewTag != null && options != null) {
+            val uiManager = UIManagerHelper.getUIManagerForReactTag(reactContext, viewTag)
+            context.runOnUiQueueThread {
+                val camera = uiManager?.resolveView(viewTag) as CKCamera
+                camera.capture(options.toHashMap(), promise)
+            }
         }
```

Comment:
If it's possible for this check to fail, then the promise should be rejected in the else branch. If not, why not just assert?